src/nsrt_learning.py

```diff
@@ -234,13 +243,18 @@ def unify_effects_and_options(
         lifted_add_effects: FrozenSet[LiftedAtom],
         ground_delete_effects: FrozenSet[GroundAtom],
         lifted_delete_effects: FrozenSet[LiftedAtom],
+        ground_param_option: ParameterizedOption,
+        lifted_param_option: ParameterizedOption,
         ground_option_args: Tuple[Object, ...],
-        lifted_option_args: Tuple[Variable, ...],
+        lifted_option_args: Tuple[Variable, ...]
 ) -> Tuple[bool, ObjToVarSub]:
     """Wrapper around utils.unify() that handles option arguments, add effects,
     and delete effects. Changes predicate names so that all are treated
     differently by utils.unify().
     """
+    # Can't unify if the parameterized options are different.
+    if ground_param_option != lifted_param_option:
```

Comment:
Note, of course, we could directly check this in the loop above. But fine to keep all the unification logic in one place, even if it's trivial as in this case.